### ISSUE TITLE
entry numbers added to some normative reference entries

### ISSUE THREAD

**Issue by ReesePlews** [metanorma/metanorma-ieee#374]
_@opoudjis i am not sure if this is my fault or a software issue. i am not sure if i should list it here (template issue) or under the relaton repo. any advice on a workaround would be greatly appreciated. thank you._

i shifted two references that were previously in my bibliography (rendering well) to the normative references clause of my document. the references were "commented out" in the bibliography so as to not be duplicated.

the generate assigned entry numbers (both [1]) to each reference in the normative reference clause; the PDF render is shown below:

![image](https://github.com/metanorma/metanorma-ieee/assets/35089343/3c51d46a-e667-4105-ad0b-403b581bdfd2)

the current code for my normative reference clause is shown here:

```
[bibliography]
== Normative references

* [[[IEC_SRD_63188_2022,IEC SRD 63188]]]
* [[[IEEE_2413_2019,IEEE Std 2413]]]
* [[[ISO_19125-1_2004,ISO 19125-1]]]
* [[[ISO_19170-1_2021,ISO 19170-1]]]
* [[[ISO_IEC_20802_1_2016,ISO/IEC 20802-1]]]
* [[[ISO_IEC_20802_2_2016,ISO/IEC 20802-2]]]
* [[[ISO_IEC_21778_2017,ISO/IEC 21778]]]
* [[[quic_rfc,IETF RFC 9000]]]
* [[[http2_rfc,IETF RFC 9113]]]
* [[[mqtt_v5,OASIS mqtt-v5.0]]]
* [[[ogc_geopose,OGC 21-056r11]]], OGC GeoPose 1.0 Data Exchange Standard
* [[[ogc_sensorweb,OGC 07-165r1]]], OGC Sensor Web Enablement: Overview And High Level Architecture
* [[[OGC_Simple_feature_1,OGC 06-103r4]]], OGC Simple Feature Access – Part 1: Common Architecture
* [[[w3c_did,W3C did-core]]], Decentralized Identifiers (DIDs) v1.0
* [[[w3c_did_spec_registries,W3C did-spec-registries]]]
* [[[w3c_json_ld,W3C json-ld11]]], A JSON-based Serialization for Linked Data
* [[[w3c_wot,W3C wot-architecture]]], Web of Things (WoT) Architecture

* [[[h3geo,1]]]
span:organization[Uber Technologies, Inc.]
span:title[H3: Hexagonal hierarchical geospatial indexing system]
span:type[website]
span:date[2024]
Available at: span:uri[https://h3geo.org]

* [[[graphql,1]]],
span:organization[Joint Development Foundation Projects, LLC].
span:type[webresource]
span:title[The GraphQL Specification Project]
span:date[October 2021],
Available at: span:uri[https://spec.graphql.org]
```

the metanorma version is:

```
Run metanorma --version
Metanorma 1.7.7
Metanorma::Cli 1.9.8
Metanorma::Standoc 2.8.9/IsoDoc 2.10.2
Metanorma::ISO 2.7.8
Metanorma::Iec 2.4.7
Metanorma::IEEE 1.2.8
Metanorma::Ietf 3.3.6
Metanorma::Generic 2.6.3
Metanorma::BIPM 2.4.7
Metanorma::CC 2.4.7
Metanorma::Csa 2.4.7
Metanorma::IHO 0.9.7
Metanorma::JIS 0.2.8
Metanorma::UN 0.12.9
Metanorma::Ogc 2.5.8
Metanorma::ITU 2.4.7
```

**Comment by ronaldtse** [comment]
Thanks for reporting @ReesePlews !

**Comment by ReesePlews** [comment]
no problem @ronaldtse; just wondering if that is something i am doing wrong.  i had talked with Nick about values for specific spans and these were from the bib (which was in my feeling, rendering ok) even if those span values could be modified a bit.

**Comment by opoudjis** [comment]
SDOs, and therefore SDO stylesheets, and therefore Metanorma, really don't expect to see documents in Normative References that are not standards, and that do not have an explicit document identifier. Giving normative effect to a website is not something typically foreseen. IEEE is exceptional in this regard, and even there, I'm not blown away by the level of detail of guidance in the IEEE style guide; it is clearly treating non-standards normative references as an afterthought. It refers out to the Chicago Manual of Style, but does that mean footnote (as the web example suggests), or author-date? 

We should deal with cases like this better than we are, but this will have been the first Metanorma flavour to which we introduce either author-date or footnote citation. We have some scaffolding supporting that in in relaton-render, but no flavour has actually required them hitherto; https://github.com/metanorma/metanorma-standoc/issues/168 has been unactioned for 5 years.

It's clear what's happening here: there is no docidentifier, the IEEE stylesheet expects a docidentifier (what IEEE calls a designator, because we currently only cater for standards in Normative References), it's not finding one, it's assigning [1] in both cases as the metanorma-ordinal default (which would kick in in a Bibliography), and the result is hilarity.

But if we follow the IEEE style guide, websites such as these do not get Normative References entries at all. They get footnotes. 2021 IEEE SA Standards Style Manual, §12.3.6.

This is a lot of work to realise non-standards normative references the way IEEE expects, and I have a lot of pressing work that isn't getting done (not least of which is Plateau), and an ongoing complete vacuum of task priority-setting. If you want this, @ReesePlews, you're going to have to give me a timeframe, and @ronaldtse you're going to have to prioritise tickets, giving the ongoing gaping absence of additional Metanorma resourcing. 

And I'm going to need someone in IEEE  Editorial to confirm that websites really do get bibliographic entries in the normative references, and (presumably) citations generated as footnotes out of cross-references—as opposed to just on-the-stop footnotes, which you can do right now. At minimum, a published IEEE standard displaying that kind of bibliographic reference to a website in its Normative References, and how it is cited; the Style Guide's vague mention of CMS is insufficient for this. We cannot undertake days of work on something that IEEE Editorial could reject as non-conformant.

**Comment by ReesePlews** [comment]
thank you @opoudjis ; thanks for the feedback.  you mention

> And I'm going to need someone in IEEE Editorial to confirm that websites really do get bibliographic entries in the normative references, and (presumably) citations generated as footnotes out of cross-references—as opposed to just on-the-stop footnotes, which you can do right now. At minimum, a published IEEE standard displaying that kind of bibliographic reference to a website in its Normative References, and how it is cited; the Style Guide's vague mention of CMS is insufficient for this. We cannot undertake days of work on something that IEEE Editorial could reject as non-conformant.

it was IEEE-SA during the editorial check that IEEE-SA told use if we used a document normatively (in case of a requirement) then we needed to add them to clause 2. it also seemed strange to me. ISO tends to make authors remove normative refs unless they are truly required and indispensable documents. 

maybe my [interpretation ](https://github.com/Spatial-Web-Foundation/SWF-Corpus_and_IEEEP2874-D2/issues/1528)of what IEEE-SA was expecting "normative refs" are is beyond what IEEE-SA is thinking. 

as for the current document, we can edit the generated word file, (or even possibly the pdf) to remove them manually, not an issue.

before getting started on this, lets check with @ronaldtse he has been in touch with IEEE-SA for another issue, so based on your reply, i now think it would be good to hear what IEEE-SA say/advise before any work/modifications happen.

thanks again for the feedback. i will let you and Ron coordinate if this should remain open or become a new issue for future work.

**Comment by opoudjis** [comment]
Most expedient thing to do here is to simply delete the ordinal identifiers from Normative References, no matter if Metanorma has inserted them, to ensure that they are never used.

**Comment by opoudjis** [comment]
Cross-references to these are currently defaulting to the anchor in the absence of an identifier. The IEEE Style Manual quite simply does not define what the designation of a non-standard normative reference is; the vague referencing of the Chicago Manual of Style implies Author-Date, but the treatment of internet sources implies footnotes. I'm not going to deal with this until it comes up, because in this instance, the source should be left as a footnote anyway.

The entries are currently being processed with a `(NO ID)` identifier in the normative references if I suppress it; I need to suppress `(NO ID)` in this flavour.

**Comment by opoudjis** [comment]
Internationalise "(NO ID)", which will allow it to be configured; it should be possible to set it to nil.

**Comment by opoudjis** [comment]
We need to change deep_merge so that nil can replace a value.

**Comment by opoudjis** [comment]
![Screenshot 2024-07-19 at 00 27 30](https://github.com/user-attachments/assets/07ec4d30-3b0c-4848-8aaf-20ee50581f01)

**Comment by ReesePlews** [comment]
@opoudjis thank you for checking and addressing this issue. the document is out of ballot and will be revised in the coming days. i will let you know if there are any issues. thank you for working on this.

**Comment by opoudjis** [comment]
No problem, @ReesePlews . I didn't want to work on it because of the somewhat opaque instructions around how to reference websites, but Ronald had convinced me that it still needed to be addressed. 

We've just got confirmation from IEEE that they expect Author-Date referencing for non-standards/websites, which will be a new ticket.

**Comment by ReesePlews** [comment]
thanks @opoudjis i will generate the document tomorrow, unless it will take longer for the docker instances to be updated.
